pkg/collector/check/stats/stats.go

```diff
@@ -249,6 +257,9 @@ func (cs *Stats) Add(t time.Duration, err error, warnings []error, metricStats S
 		cs.TotalEventPlatformEvents[k] = cs.TotalEventPlatformEvents[k] + v
 		cs.EventPlatformEvents[k] = v
 	}
+	if haagent != nil && haagent.Enabled() && haagent.IsHaIntegration(cs.CheckName) {
```

Comment:
Do you think we could include the `haagent.Enabled()` check in `IsHaIntegration` to simplify usage ?

Reply:
Hum, I think it's not ideal since they mean different things. Can be misleading to have `haagent.Enabled()` check in `IsHaIntegration()` since a check can be flagged as HA, but HA Agent feature can be disabled.